### PR TITLE
ADR - Forms states and using AASM for state machine

### PR DESCRIPTION
**Pull request by aliuk2012**
# PR Checklist

- [ ] If you are proposing a new decision record document, used the right template for that
      - ([ADR](https://github.com/alphagov/forms/blob/main/ADR/ADRXXX-architecture-decision-record-template.md), [decision-record](https://github.com/alphagov/forms/blob/main/decision-record/DRXXX-decision-record-template.md), engagement, [research](https://github.com/alphagov/forms/blob/main/research/YYYY-MM-DD-template.md))
- [ ] Set yourself as the Assignee
- [ ] Tag anyone you would like to review, or @forms-design or @forms-devs
- [ ] Fill in the template below

---

# What

Describe what you have changed and why.

# How to review

Describe the steps required to test the changes.

For example:

1. Semantic: Do you agree with the changes?
1. Syntactic: Spelling, grammar, etc.

# Who can review

Describe who can review the changes. Or more importantly, list the people
that can't review, because they worked on it.

### CODE REVIEW

File: ADR/ADR025-use-aasm-to-manage-form-state.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+> Accepted
+
+## Context
+
+### Before
+
+Initially, there only three "states" that a form could be in, "Draft", "Live" and a combination of the two "Live with draft". These states were not hardcoded and we inferred by a combination of attribute  
+
+1. Draft - All "Form" records are draft
```
Comment (thomasiles) [inline ADR/ADR025-use-aasm-to-manage-form-state.md:15]:
```suggestion
1. Draft - All "Form" records start as in draft
```

File: ADR/ADR025-use-aasm-to-manage-form-state.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+introduce new states like `archived` forms.
+
+The follow work was done in this order to add proper state management:
+
+- [add state column to form model and bare bones state machine](https://github.com/alphagov/forms-api/pull/422)
+- [tidy up migrations into a single migration file](https://github.com/alphagov/forms-api/pull/428)
+- [Integrate FormStateMachine into Form model and start tracking state through events](https://github.com/alphagov/forms-api/pull/425)
+- [Change has_draft_version & has_live_version to use form state column](https://github.com/alphagov/forms-api/pull/429)
+
+While implementing the above, we discovered we needed to temporaryly introduce a new state to transistion from `live` to `draft`
```
Comment (thomasiles) [inline ADR/ADR025-use-aasm-to-manage-form-state.md:61]:
```suggestion
While implementing the above, we discovered we needed to temporarily introduce a new state to transition from `live` to `draft`
```

**Review by thomasiles — APPROVED** [review]
I think this documents things well, nice one. I spotted a couple of typos but I'm not certain and I don't think it matters.

File: ADR/ADR025-use-aasm-to-manage-form-state.md (hunk 1 above)
Comment (thomasiles) [reply ADR/ADR025-use-aasm-to-manage-form-state.md:15]: I think I've made this less clear 🤦

Comment (aliuk2012) [reply ADR/ADR025-use-aasm-to-manage-form-state.md:15]: What I was trying to say is that when looking at `Form` record before AASM was introduced, we should consider `Form` as being a draft record which a user can also edit at any point but it wouldn't be made live until a snapshot of it has been copied over to made_live_forms

**Review by thomasiles — APPROVED** [review]

**Pull request merged**